performance/src/efficiency/finding_performance_writer.cpp

```diff
@@ -126,22 +156,43 @@ void finding_performance_writer::write_common(
         const std::vector<measurement>& measurements = tracks[i];
 
         // Check which particle matches this seed.
+        // Input :
+        //    - the list of measurements for this track
+        //    - the truth particles map
+        // Output :
+        //    - a list of particles, having for each of them a particle_id and
+        //      a count value.
+        // If there is only a single truth particle contributing to this track,
+        // then increment the match_counter for this truth particle id.
+        // If there are at least two particles contributing to the hit list of
+        // this track, increment the fake_counter for each truth particle.
+
         std::vector<particle_hit_count> particle_hit_counts =
             identify_contributing_particles(measurements, evt_map.meas_ptc_map);
 
         if (particle_hit_counts.size() == 1) {
             auto pid = particle_hit_counts.at(0).ptc.particle_id;
             match_counter[pid]++;
         }
+
+        if (particle_hit_counts.size() > 1) {
+            for (particle_hit_count const& phc : particle_hit_counts) {
```

Comment:
Can you explain the detail of logic? If there are more than one particle contributing to the track, it is considered to be a fake track?

According to this definition and the later changes below, I guess one single track can increase the fake count multiple times? (i.e. If there are three particles contributing to a track then the fake count increases by 3?)

Reply:
Thanks for your review :)

I defined a `fake track` as “a track made with measurements from more than one single truth particle”.

For each `truth particle`, the `fake rate` tells how many `fake tracks` were reconstructed with at least one measurement belonging to this particle.

The ROOT plots show “the number of `fake tracks` per particle property/value”, so it’s the other way around if I understood you correctly. It’s: if 3 `fake tracks` share one common particle, then the `fake count` for this particle will be 3.

If a `fake track` is made of measurements belonging to 3 truth particles, then the `fake_counter` for each particle will be increased by 1.

Reply:
Thanks for the explanation. Then the y-axis of the fake rate plots is the averaged number of fake tracks per particle?

Reply:
Sorry for taking so long to reply. Yes, absolutely, I have just changed the plot title to `"Averaged number of fake tracks per particle"`, that's a much better and informative title, thanks!